### PR TITLE
[feat] support overwriting of state

### LINKED ISSUE #11: Add support to reset a store
In certain use-cases I want to reset my store to a certain state

example:

```
const store = makeStore<T>({});

store.set({hello: "world"});

...

store.set({}); // does not modify the store
store.set(Object.keys(store.state).reduce((r,c) => {...r, [c]: undefined}, {})) // actually resets but is cumbersome
```

idea: 
```
store.set({}, {overwrite: true});
```

I'd be willing to create a PR if you think this a feature that should exist in `statery`

### PR DESCRIPTION
**Pull request by jaecktec**
closes #11 

- Added overwrite method that deletes all undefined values + overwrites with given state
- changed extra-options to be an object

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -95,38 +104,45 @@ export const makeStore = <T extends State>(initialState: T): Store<T> => {
...
 
-  return {
-    get state() {
-      return state
-    },
-
-    set: (incoming, forceNotify = false) => {
-      /* If the argument is a function, run it */
-      const incomingState = incoming instanceof Function ? incoming(state) : incoming
+  const _set: Store<T>["set"] = (incoming, options) => {
```
Comment (hmans) [inline src/index.ts]: Is there a specific reason for this renaming?

File: src/index.ts [hunk 2]
```diff
@@ -164,24 +180,6 @@ export const useStore = <T extends State>(store: Store<T>): T => {
...
-    if (store.state === initialState) return
-
-    subscribedProps.forEach((prop) => {
-      if (initialState[prop] !== store.state[prop]) {
-        setVersion((v) => v + 1)
-        return
-      }
-    })
-  }, [store])
-
```
Comment (hmans) [inline src/index.ts:189]: I'm surprised this entire section was removed. Was there a specific reason for this?

**Review by hmans — CHANGES_REQUESTED** [review]
Hi and sorry for taking so long to get back to this PR. I think the feature itself is fine, but I have some comments about the implementation. Thanks!

File: src/index.ts (hunk 2 above)
Comment (jaecktec) [reply src/index.ts:189]: I think this is just due to formatting, let me update the commit with a proper formatter. The section still exists

I just moved the inline definition of set to a const _set so I can use it in the overwrite method

File: src/index.ts (hunk 1 above)
Comment (jaecktec) [reply src/index.ts]: yes, so I put it to a separate variable, so accessing it does not depend on the order of the variables. I could also inline it in the return statement, but I felt it's more clean that way

**Comment by jaecktec** [comment]
Hey, sorry for the long response time, too. I was on vacation and din't touch the PC. 

I will reformat everything and update the docs. 
Also I'd undo the changes to the options object - since they are not scope of the PR.

**Comment by jaecktec** [comment]
ok, so why I moved `set` to `const _set = ..` :
I wanted to reuse the `set` method inside overwrite, and that's not possible if it's defined inline in the object

**Comment by jaecktec** [comment]
Was my feedback not appropriate?

**Comment by hmans** [comment]
I'm sorry about only getting back to you know, but I have been and still am extremely busy with my other projects, and can't currently perform the context switch required to deeply jump into this topic. I'll try to jump on top of this some time soon. Thank you very much for your patience.

**Pull request closed without merging**